src/main/java/hu/oe/nik/szfmv/detector/classes/Detector.java

```diff
@@ -0,0 +1,32 @@
+package hu.oe.nik.szfmv.detector.classes;
+
+import hu.oe.nik.szfmv.detector.interfaces.ICamera;
+import hu.oe.nik.szfmv.detector.interfaces.IRadarUltrasonic;
+import hu.oe.nik.szfmv.environment.WorldObject;
+import hu.oe.nik.szfmv.environment.models.Collidable;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Class Responsible for Sensor Functionality
+ */
+public class Detector implements ICamera, IRadarUltrasonic {
+
+    private List<WorldObject> worldObjects;
+
+    public Detector(List<WorldObject> worldObjects) {
+        this.worldObjects = worldObjects;
+    }
+
+    @Override
+    public List<WorldObject> getWorldObjects(Point a, Point b, Point c) {
+        return null;
```

Comment:
Azért nem szeretem ezt az API tervezési megoldást, mert rengeteg hibát szül. Ha egy lista típus a visszatérési érték és null-t dobsz, akkor az aki ezt a metódust használja kénytelen egy null check if-el bonyolítani a kódját. Ha egy üres listát adsz vissza, akkor a for(each) lekezeli az üres esetet implicit (üres, ergo nem hajtja végre a ciklus törzset).